contracts/src/v0.8/functions/v1_3_0_zksync/FunctionsRouter.sol

```diff
@@ -0,0 +1,584 @@
+    );
+
+    emit RequestProcessed({
+      requestId: commitment.requestId,
+      subscriptionId: commitment.subscriptionId,
+      totalCostJuels: receipt.totalCostJuels,
+      transmitter: transmitter,
+      resultCode: resultCode,
+      response: response,
+      err: err,
+      callbackReturnData: result.returnData
+    });
+
+    return (resultCode, receipt.callbackGasCostJuels);
+  }
+
+  function _callback(
+    bytes32 requestId,
+    bytes memory response,
+    bytes memory err,
+    uint32 callbackGasLimit,
+    address client
+  ) private returns (CallbackResult memory) {
+    bool destinationNoLongerExists;
+    assembly {
+      // solidity calls check that a contract actually exists at the destination, so we do the same
+      destinationNoLongerExists := iszero(extcodesize(client))
+    }
+    if (destinationNoLongerExists) {
+      // Return without attempting callback
+      // The subscription will still be charged to reimburse transmitter's gas overhead
+      return CallbackResult({success: false, gasUsed: 0, returnData: new bytes(0)});
+    }
+
+    bytes memory encodedCallback = abi.encodeWithSelector(
+      s_config.handleOracleFulfillmentSelector,
+      requestId,
+      response,
+      err
+    );
+
+    uint16 gasForCallExactCheck = s_config.gasForCallExactCheck;
+
+    // Call with explicitly the amount of callback gas requested
+    // Important to not let them exhaust the gas budget and avoid payment.
+    // NOTE: that callWithExactGas will revert if we do not have sufficient gas
+    // to give the callee their requested amount.
+
+    bool success;
+    uint256 gasUsed;
+    uint256 g1 = gasleft();
+
+    assembly {
+      let g := gas()
+      // Compute g -= gasForCallExactCheck and check for underflow
+      // The gas actually passed to the callee is _min(gasAmount, 63//64*gas available).
+      // We want to ensure that we revert if gasAmount >  63//64*gas available
+      // as we do not want to provide them with less, however that check itself costs
+      // gas. gasForCallExactCheck ensures we have at least enough gas to be able
+      // to revert if gasAmount >  63//64*gas available.
+      if lt(g, gasForCallExactCheck) {
+        revert(0, 0)
+      }
+      g := sub(g, gasForCallExactCheck)
+      // if g - g//64 <= gasAmount, revert
+      // (we subtract g//64 because of EIP-150)
+      if iszero(gt(sub(g, div(g, 64)), callbackGasLimit)) {
+        revert(0, 0)
+      }
+    }
+
+    // allocate return data memory ahead of time
+    bytes memory returnData = new bytes(MAX_CALLBACK_RETURN_BYTES);
+
+    // solhint-disable-next-line avoid-low-level-calls
+    (success, returnData) = GAS_BOUND_CALLER.delegatecall{gas: callbackGasLimit}(
+      abi.encodeWithSelector(IGasBoundCaller.gasBoundCall.selector, client, callbackGasLimit, encodedCallback)
+    );
+
+    uint256 pubdataGasSpent;
+    if (success) {
+      (, pubdataGasSpent) = abi.decode(returnData, (bytes, uint256));
```

Comment:
i understand that we want to limit the return data size to `MAX_CALLBACK_RETURN_BYTES`. This limit is pretty small? i am not sure if it's a case for revert or we can continue the execution. 

if the delegatecall is successful but the return data size is too big, will that make `pubdataGasSpent` unpredictable?